team_check.py

```diff
@@ -0,0 +1,78 @@
+import pandas as pd
+from github3 import login
+import configparser
+import numpy as np
+
+
+### Get team members
+config = configparser.ConfigParser()
+config.read('config')
+config = config['DEFAULT']
+
+username = config['username']
+token = config['token']
+organization = config['organization']
+pairs = config['pairs']
+roster = config['roster']
+
+with open(token, 'r') as f:
+    token = f.readline().strip()
+    
+github = login(username,token=token)
+memberships = github.organization_memberships()
+course = None
+
+for membership in memberships:
+    if membership.organization.login.lower() == organization.lower():
+        course = membership.organization
+        break
+    
+teams = {team.name : team for team in list(course.teams())}
+tas = teams['TAs'].members()
+instructors = teams['Instructors'].members()
+print("Getting team members, please wait...")
+team_members = {team.name: [m.login for m in set(team.members()) - \
+    set(tas) - set(instructors)] for team in teams.values()}
+
+
+### Preprocess
+pairs_df = pd.read_csv(pairs)
+roster_df = pd.read_csv(roster)
+
+header = ['CCID', 'GitHub username', 'expected team', 'current team']
+pairs_df = pairs_df[['Email', 'Assignment 1 Team']]
+pairs_df.columns = ['CCID', 'expected team']
+pairs_df = pairs_df.reindex(columns = header)
+
+roster_df = roster_df[['github_username', 'identifier', 'group_name']]
+roster_df.columns = ['GitHub username', 'CCID', 'expected team']
+
+
+### Check student teams
+pairs_result = pairs_df.copy()
+pairs_result['GitHub username'] = pd.merge(pairs_df, roster_df, on='CCID')['GitHub username_y']
+
+pairs_result['current team'] = pairs_result['GitHub username'].apply(lambda x: \
+    [k for k,v in team_members.items() if x in v]).apply(lambda x: len(x) \
+        and (len(x) == 1 and x[0] or str(x)) or np.nan)
+
+pairs_result = pairs_result.replace(np.nan, '')
+pairs_result = pairs_result.loc[pairs_result['expected team'].str[-3:] != \
+    pairs_result['current team'].str[-3:]].reset_index(drop=True)
```

Comment:
Tons of students named their team wrongly, e.g., `Agmt1_Team20` should be `Asgmt1_Team20`.
To make things easier, this script simply compares the last 3 characters.

Reply:
but if they named their teams incorrectly, this will cause an issue for us to look them up? 
Actually, I thought about something for marking. Right now, each TA has a list of CCIDs to mark, but in the assignment, since students work in pairs, it makes sense to give each TA a list of pairs to mark. This means we have to update the get repos script to accept either a list of CCIDs or a list of "postfixes", the pair names in teh case of the assignments (sorry that this is a bit of a nightmare; it's the first term we do pair assignments in)

Reply:
This can be handled by doing the same thing: 
in collect_repos.py, if the prefix is assignment, it only checks the last three characters
or
we need to rename all these incorrect teams on GitHub

Reply:
hmm so there are two separate issues here:
(1) from what I understand, if a list of ccids are given, collect_repos.py will map these ccids to the corresponding github usernames and only clone/update the respective repos. If no ccid list is given, it will clone all repos related to this assignment prefix. The current behavior does not support providing a list of assignment teams (i.e., really the equivalent of repo name postfixes).

(2) the incorrect team names on GitHub.  For that, I think checking the last three characters as you suggest should be enough, but this would also need to be done for the grade mapping/generation (probably the latter is in the marking scripts not this repo).

Reply:
In collect_repos.py, it maps ccids to GitHub usernames using a mapping read from `roster.csv`.
So maybe we can do this: if the prefix is assignment, read a mapping from `pairs.csv`. otherwise, read from `roster.csv`

Reply:
The issue with this solution is that then two TAs will then mark the same team (partner 1 ccid is with TA1 and partner 2 ccid is with TA2 so they both mark the same repo)

Reply:
If the prefix is assignment, we simply don't do the mapping?
we put team names in ccid_list, then the script pulls all repos with the last 3 chars matching.
The script will look jumbled though.

Reply:
My suggestion would be:

- In this repo, have another optional file called `repo_names.txt` which contains the team names a TA wants to grade. Basically, this is the suffix of the assignment name. So this list can be either the student's github username (for individual things like labs-X) or the team name for group work like (assignment1-X). If the script sees that this file is supplied, then it doesn't look at the ccid_list.
- Let us fix the incorrect names on github such that the script here just looks up the exact thing so it's not super customized to the last 3 chars matching and doesn't make the script look jumbled. Do you have the incorrect team names? I can fix them and email their team members.

Reply:
Just to be clear: we should not change the current ccid_list logic..the team names is an alternative

Reply:
Here's the list:
https://drive.google.com/file/d/1xbYNOfQ2fTqwaU2-aXbBSZqi3oGi4CKV/view?usp=sharing
not as many as I though

Reply:
Ok, I'll fix these now and email the students.